src/backend/distributed/README.md

```diff
@@ -1749,6 +1749,60 @@ The reason for handling dependencies and deparsing in post-process step is that
 
 Not all table DDL is currently deparsed. In that case, the original command sent by the client is used. That is a shortcoming in our DDL logic that causes user-facing issues and should be addressed. We do not directly construct a separate DDL command for each shard. Instead, we call the `worker_apply_shard_ddl_command(shardid bigint, ddl_command text)` function which parses the DDL command, replaces the table names with shard names in the parse tree according to the shard ID, and then executes the command. That also has some shortcomings, because we cannot support more complex DDL commands in this manner (e.g. adding multiple foreign keys). Ideally, all DDL would be deparsed, and for table DDL the deparsed query string would have shard names, similar to regular queries.
 
+`markDistributed` is used to indicate whether we add a record to `pg_dist_object` to mark the object as "distributed".
+
+## Defining a new DDL command
+
+All commands that are propagated by Citus should be defined in DistributeObjectOps struct. Below is a sample DistributeObjectOps for ALTER DATABASE command that is defined in [distribute_object_ops.c](commands/distribute_object_ops.c) file.
+
+```c
+static DistributeObjectOps Database_Alter = {
+	.deparse = DeparseAlterDatabaseStmt,
+	.qualify = NULL,
+	.preprocess = PreprocessAlterDatabaseStmt,
+	.postprocess = NULL,
+	.objectType = OBJECT_DATABASE,
+	.operationType = DIST_OPS_ALTER,
+	.address = NULL,
+	.markDistributed = false,
+};
+```
+
+Each field in the struct is documented in the comments within the `DistributeObjectOps`. When defining a new DDL command, follow these guidelines:
+
+- **Returning tasks for `preprocess` and `postprocess`**: Ensure that either `preprocess` or `postprocess` returns a list of "DDLJob"s. If both functions return non-empty lists, then you would get an assertion failure.
+
+- **Generic `preprocess` and `postprocess` methods**: ``PreprocessAlterDistributedObjectStmt`` and ``PostprocessAlterDistributedObjectStmt`` are generic post and pre methods that is being used for some statements. Both of them are being used for distributed object operations.
+
+	- PreprocessAlterDistributedObjectStmt perform below operations perform qualify operation, deparse the statement and generate a task list. PostprocessAlterDistributedObjectStmt - calls ``EnsureAllObjectDependenciesExistOnAllNodes`` to propagate missing dependencies both on coordinator and worker.
```

Comment:
This is useful, but it's hard to read. I think formatting it like this makes it easier to understand:
- `PreprocessAlterDistributedObjectStmt` performs the below operations:
   1. perform qualify operation
   2. deparse the statement
   3. generate a task list
- `PostprocessAlterDistributedObjectStmt` instead does:
   1. calls `EnsureAllObjectDependenciesExistOnAllNodes` to propagate missing dependencies both on coordinator and worker.

Reply:
Fixed the format and rephrased the text to make it more clear